src/boosting/score_updater.hpp

```diff
@@ -34,9 +34,13 @@ class ScoreUpdater {
     const double* init_score = data->metadata().init_score();
     // if exists initial score, will start from it
     if (init_score != nullptr) {
-      if ((data->metadata().num_init_score() % num_data_) != 0
-          || (data->metadata().num_init_score() / num_data_) != num_tree_per_iteration) {
-        Log::Fatal("Number of class for initial score error");
+      if ((data->metadata().num_init_score() % num_data_) != 0) {
+        Log::Fatal("Number of class for initial score error. Score count is %d. num_data is %d.",
+          data->metadata().num_init_score(), num_data_);
+      }
+      if ((data->metadata().num_init_score() / num_data_) != num_tree_per_iteration) {
+        Log::Fatal("Number of class for initial score error.  Found %d classes.  Should be %d",
+          (data->metadata().num_init_score() / num_data_), num_tree_per_iteration);
```

Comment:
I'm ok with splitting this into two different error messages to provide a bit more specific hint about what went wrong. Thanks for the suggestion!

Since this change seems unrelated to the feature "add streaming APIs", could you split it off into a separate PR? That would allow us to review and merge it pretty quickly, would reduce the size of this PR, and would help maintainers to understand the impact of the proposed changes in this PR (e.g. we'd be able to say "ok, this PR didn't have to touch `score_updater` at all").